### DISCUSSION TITLE
Display not suspending after upgrade to 22.10/Regolith 2.2

### DISCUSSION THREAD

**Discussion by stephenh** [regolith-linux/regolith-desktop#804] [Q&A]
After `do-release-upgrade`-ing from 22.04 to 22.10 on two machines, neither machine's display is going into standby after ~10-15m of inactivity.

For both machines, I did the `sudo apt install regolith-i3-control-center-gnome` fix, so can open up the control center now, and both have "Screen Blank" set to 15m:

![image](https://user-images.githubusercontent.com/6401/210251046-9dd3ffb2-120b-4b06-93a4-0c218f40523f.png)

However it doesn't actually work. It seems like DPMS is not configured in X:

```
$ xset q
...
DPMS (Energy Star):
  Standby: 0    Suspend: 0    Off: 0
  DPMS is Enabled
  Monitor is On
```

I can manually force standby by running `xset dpms force standby` but would like to get auto-standby working again.

Are there any ideas for way the control center setting would not show up in `xset q`? Should it? Or is there some other process/config setting that controls DPMS?

(I.e. while researching fixes, I read that xscreensaver has its own standby/DPMS config, but I'm not using that; I guess I do have `xdg-screensaver` installed (I did not install it manually) and `xdg-screensaver status` says enabled; which, technically I think the screensaver/lock screen is not working either, but I generally care less about than than standby. ... note that I don't have `i3-wm.program.lock` set and have generally/iirc never touched my lock/screensaver/standup config, I've always just used the default + using control center to set the screen blank time.)

It also looks like `gnome-flashback` is enabled: 

```
$ gsettings get org.gnome.gnome-flashback screensaver
true
```

I will keep poking around, but any tips/pointers/fixes would be appreciated. Thanks!

**Comment by kgilmer** [comment]
To isolate the regression to Regolith or `gnome-flashback`, are you able to install the `gnome-session-flashback` package and log in to that?  If it also shows the issue than the problem is upstream, if not something in Regolith.

**Reply by stephenh** [reply]
Huh! So, on one machine I did:

* Install `gnome-session-flashback`, logged into metacity, confirmed screen suspends after 1m (sweet!)
* Logged into regolith instead of metacity, and confirmed screen suspends after 1m
* Removed `gnome-session-flashback` and it "removed too much", and I couldn't boot to a login manager * Installed `regolith-system-ubuntu` and that fixed booting to a login manager (whew!)
* Now I can login to regolith, and screen suspends after 1m, so it works!

So on the other machine, I merely did:

* Installed `gnome-session-flashback`
* Rebooted directly into regolith/i3, and
* Confirmed the screen now suspends after 1m, w/o ever having logged into metacity

So simply installing `gnome-session-flashback` seems to have fixed it.

Here is what `apt install gnome-session-flashback` brought into my 2nd machine:

```
[I] ~ $ sudo apt install gnome-session-flashback
[sudo] password for stephen:
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following package was automatically installed and is no longer required:
  libavahi-ui-gtk3-0
Use 'sudo apt autoremove' to remove it.
The following additional packages will be installed:
  accountsservice-ubuntu-schemas alacarte bamfdaemon evolution-data-server evolution-data-server-common gnome-applets gnome-applets-data gnome-panel gnome-panel-data indicator-applet indicator-application
  indicator-appmenu indicator-bluetooth indicator-common indicator-datetime indicator-keyboard indicator-messages indicator-power indicator-printers indicator-session indicator-sound jayatana libabsl20210324
  libaccounts-glib0 libbamf3-2 libcamel-1.2-64 libcpufreq0 libebackend-1.2-11 libebook-1.2-21 libebook-contacts-1.2-4 libecal-2.0-2 libedata-book-1.2-27 libedata-cal-2.0-2 libedataserver-1.2-27 libedataserverui-1.2-4
  libfcitx-gclient1 libfcitx-utils0 libgucharmap-2-90-7 libido3-0.1-0 libindicator3-7 libmessaging-menu0 libmetacity3 libphonenumber8 libprotobuf23 libunity-gtk2-parser0 libunity-gtk3-parser0 libxpresent1 metacity
  metacity-common ubuntu-touch-sounds unity-gtk-module-common unity-gtk2-module unity-gtk3-module unity-settings-daemon-schemas
Suggested packages:
  evolution gnote evolution-common compiz unity-greeter-session-broadcast fcitx
Recommended packages:
  appmenu-qt
The following packages will be REMOVED:
  regolith-system-ubuntu
The following NEW packages will be installed:
  accountsservice-ubuntu-schemas alacarte bamfdaemon evolution-data-server evolution-data-server-common gnome-applets gnome-applets-data gnome-panel gnome-panel-data gnome-session-flashback indicator-applet
  indicator-application indicator-appmenu indicator-bluetooth indicator-common indicator-datetime indicator-keyboard indicator-messages indicator-power indicator-printers indicator-session indicator-sound jayatana
  libabsl20210324 libaccounts-glib0 libbamf3-2 libcamel-1.2-64 libcpufreq0 libebackend-1.2-11 libebook-1.2-21 libebook-contacts-1.2-4 libecal-2.0-2 libedata-book-1.2-27 libedata-cal-2.0-2 libedataserver-1.2-27
  libedataserverui-1.2-4 libfcitx-gclient1 libfcitx-utils0 libgucharmap-2-90-7 libido3-0.1-0 libindicator3-7 libmessaging-menu0 libmetacity3 libphonenumber8 libprotobuf23 libunity-gtk2-parser0 libunity-gtk3-parser0
  libxpresent1 metacity metacity-common ubuntu-touch-sounds unity-gtk-module-common unity-gtk2-module unity-gtk3-module unity-settings-daemon-schemas
0 upgraded, 55 newly installed, 1 to remove and 8 not upgraded.
```

Do you recognize anything in there that would nudge suspend-after-inactivity to work?

I guess I'm a little confused, b/c in theory I `apt remove`-d and `autoremove`-d everything related to `gnome-session-flashback` on 1st first machine, so effectively that I couldn't boot to a login manager, and yet after installing `regolith-system-ubuntu`, the "make screen suspend work" seems to have stuck. So unless a package stuck around, I'm not sure I really have any net-new packages on my 1st machine, but going through the install/uninstall just made it work.

I suppose it's not that important, b/c the issue is fixed for me; granted, I've not yet uninstalled `gnome-session-flashback` and reinstalled `regolith-system-ubuntu` on my 2nd machine, but I'm assuming it will have the same effect of having suspend-after-inactivity continue to work.

Note that, even now with suspend working, the `xset q` output still shows `suspend: 0`:

```
$ xset q | grep Stand
  Standby: 0    Suspend: 0    Off: 0
```

So my random googling was a miss wrt that being part of the issue. :shrug: 

Happy to close this out as just weird upgrade drift; both machines I'm pretty sure I installed from Regolith ISO 1.6 and have been upgrading since. ...maybe Regolith 2.0...but probably 1.6, not sure.

Thanks!

**Comment by kgilmer** [comment]
@stephenh thank you for all the details, very helpful.  My guess is that the package providing the functionality is `metacity-common`.  If you have time, could you see if the regression returns without that package?

**Reply by stephenh** [reply]
Hi @kgilmer , sure I'll try that.

Hrm, I removed `metacity-common`, which also removed `gnome-session-flashback`, and after rebooting, it turns out suspend-after-1m still works. 

Which seems good in the grand schema of things, but doesn't help pinpoint what the root cause of my original issue was.

Happy to try something else if you have other ideas, but also happy to chalk it up as a fluke. Thanks!

**Comment by giomf** [comment]
Hello

I have the same problem. I am running on debian sid with the unstable/testing version of Regolith. If I can contribute any information then feel free to let me know.

**Comment by neilbags** [comment]
Ok I believe I've figured this out. The missing component is
`/usr/libexec/gnome-flashback-idle-monitor` 

Although this is installed with regolith via the dependency on `gnome-flashback`, the autostart file is missing:
`/etc/xdg/autostart/gnome-flashback-idle-monitor.desktop`

This is part of the `gnome-session-flashback` package. So this package should be made a dependency, or an autostart for this daemon should be included through another mechanism. Should I open a bug report for this or is this discussion comment sufficient?

**Reply by stephenh** [reply]
Ah wow! I have two regolith machines:

* The one with `gnome-flashback-idle-monitor.desktop` _does_ auto-suspend successfully
* The one _without_ `gnome-flashback-idle-monitor.desktop` _does not_ auto-suspend successful

So @neilbags 's assertion lines up with my data points. Thank you!
